src/int64_util.js

```diff
@@ -17,85 +17,112 @@
  * under the License.
  */
 
-// This file is copied from the Thrift project under Apache 2.0 license. There are some changes to variable names and
-// styling. The other changes are documented before the changes
+// This file is copied from the Thrift project under the Apache 2.0 license. There are some changes to variable names,
+// styling and code.
 
 var Int64 = require('node-int64');
 
 var Int64Util = module.exports = {};
 
+var BYTE_BUFFER_SIZE = 8;
+
+var BYTE_7 = 7;
+var BYTE_6 = 6;
+var BYTE_5 = 5;
+var BYTE_4 = 4;
+var BYTE_3 = 3;
+var BYTE_2 = 2;
+var BYTE_1 = 1;
+
+var MAX_NUMBER_OF_DECIMAL_DIGITS_REPRESENTABLE_BY_DOUBLE = 15;
+var NUMBER_OF_LEAST_SIGNIFICANT_DIGITS = 11;
+
 var POW2_24 = Math.pow(2, 24);
 var POW2_31 = Math.pow(2, 31);
 var POW2_32 = Math.pow(2, 32);
-var POW10_11 = Math.pow(10, 11);
+var POW2_48 = Math.pow(2, 48);
+var POW10_11 = Math.pow(10, NUMBER_OF_LEAST_SIGNIFICANT_DIGITS);
+var POW10_15 = Math.pow(10, MAX_NUMBER_OF_DECIMAL_DIGITS_REPRESENTABLE_BY_DOUBLE);
+var POW2_48_OVER_POW10_11 = Math.floor(POW2_48 / POW10_11);
+var POW2_48_REMAINDER_POW10_11 = Math.floor(POW2_48 % POW10_11);
+var POW10_15_OVER_POW2_32 = Math.floor(POW10_15 / POW2_32);
+var POW10_15_REMAINDER_POW2_32 = Math.floor(POW10_15 % POW2_32);
+
+var FOUR_BYTES_BITMASK = 0xffffffff;
+var FOUR_BYTES_HIGH_1_BITMASK = 0x80000000;
+var ONE_BYTE_BITMASK = 0xff;
+var ONE_BYTE_HIGH_1_BITMASK = 0x80;
+var ONE_BYTE_HIGH_3_BITMASK = 0xe0;
+
+var ONE_BYTE_BIT_LENGTH = 8;
+var TWO_BYTES_BIT_LENGTH = 16;
+
+var MAX_NUMBER_OF_CHARACTERS_FOR_INT64 = 19;
+var MIN_NUMBER_OF_CHARACTERS_FOR_INT64 = 16;
 
 Int64Util.toDecimalString = function (i64)
```

Comment:
do we have unit tests for all these functions?

Reply:
yes, we do